### PR TITLE
maven_artifact.py - add support for version ranges by using spec (#54…

### PR DESCRIPTION
**Pull request by pjrm**
##### SUMMARY
This PR allows to specify which versions of maven packages we want to download by using constraints.
The simple use case is when we want to download all patch versions released, but making sure that we don't download a major one.

Also I saw that other person have asked for this feature on [#20003](https://github.com/ansible/ansible/issues/20003)

The syntax supported for this module was based on this [page](https://cwiki.apache.org/confluence/display/MAVENOLD/Dependency+Mediation+and+Conflict+Resolution#DependencyMediationandConflictResolution-DependencyVersionRanges).

This is the fix of reverted commit on https://github.com/ansible/ansible/pull/61488 

##### ISSUE TYPE
Feature Pull Request

##### COMPONENT NAME
maven_artifact.py

### CODE REVIEW

**Comment by turb** [comment]
shipit

**Comment by tumbl3w33d** [comment]
shipit

**Comment by turb** [comment]
shipit

**Comment by tumbl3w33d** [comment]
shipit

File: test/sanity/ignore.txt [hunk 1]
```diff
@@ -4889,7 +4889,6 @@ lib/ansible/modules/packaging/language/easy_install.py validate-modules:doc-defa
 lib/ansible/modules/packaging/language/easy_install.py validate-modules:parameter-type-not-in-doc
 lib/ansible/modules/packaging/language/easy_install.py validate-modules:doc-missing-type
 lib/ansible/modules/packaging/language/gem.py validate-modules:parameter-type-not-in-doc
-lib/ansible/modules/packaging/language/maven_artifact.py validate-modules:doc-default-does-not-match-spec
```
Comment (gundalow) [inline test/sanity/ignore.txt:4892]: Thank you for fixing this.

**Comment by gundalow** [comment]
@pjrm Thank you for this update
@turb @tumbl3w33d Thank you for your reviews.

Merged into `devel` for release in Ansible 2.10

**Pull request merged**